conan/tools/microsoft/msbuild.py

```diff
@@ -48,11 +47,75 @@ def command(self, sln, targets=None):
 
         return cmd
 
-    def build(self, sln, targets=None):
-        cmd = self.command(sln, targets=targets)
+    def build(self, sln, targets=None, force_import_generated_files=False):
+        cmd = self.command(sln, targets=targets, force_import_generated_files=force_import_generated_files)
         self._conanfile.run(cmd)
 
     @staticmethod
     def get_version(_):
         return NotImplementedError("get_version() method is not supported in MSBuild "
                                    "toolchain helper")
+
+    def _get_concrete_props_file(self, root_props_file):
+        concrete_props_file = ""
+
+        root = ET.parse(root_props_file).getroot()
+        importgroup_element = root.find("ImportGroup")
+        if importgroup_element:
+            expected_condition = f"'$(Configuration)' == '{self.configuration}' And '$(Platform)' == '{self.platform}'"
+            for import_element in importgroup_element.iter("Import"):
+                condition = import_element.attrib.get("Condition")
+                if expected_condition == condition:
+                    concrete_props_file = import_element.attrib.get("Project")
+                    break
+
+        if concrete_props_file:
+            concrete_props_file = os.path.join(self._conanfile.generators_folder, concrete_props_file)
+
+        if not concrete_props_file or not os.path.exists(concrete_props_file):
+            raise ConanException(
+                f"MSBuildToolchain props file is missing for configuration={self.configuration} and "
+                f"platform={self.platform}. 'configuration' and 'platform' attributes must be the same "
+                "respectively in MSBuildToolchain and MSBuild."
+            )
+
+        return concrete_props_file
+
+    def _get_msbuildtoolchain_properties(self, root_props_file):
+        properties = {}
+
+        # Get properties from props file of configuration and platform
+        concrete_props_file = self._get_concrete_props_file(root_props_file)
```

Comment:
Maybe this can be avoided with ``ForceImportAfterCppTargets``?

Reply:
Yes maybe, I can try

Reply:
Nop, it doesn't help :(